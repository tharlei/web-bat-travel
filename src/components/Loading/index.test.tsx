import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import { Loading } from '.';

describe('Loading component', () => {
  it('should have animate spin class', () => {
    const { container } = render(<Loading />);

    const icon = container.querySelector('.animate-spin');
    expect(icon).toBeDefined();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
