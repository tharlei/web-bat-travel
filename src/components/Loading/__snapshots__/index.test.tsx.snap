// Vitest Snapshot v1

exports[`Loading component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-center"
  >
    <svg
      class="animate-spin font-bold text-2xl text-cyan text-center"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <desc />
      <path
        d="M0 0h24v24H0z"
        fill="none"
        stroke="none"
      />
      <line
        x1="12"
        x2="12"
        y1="6"
        y2="3"
      />
      <line
        x1="16.25"
        x2="18.4"
        y1="7.75"
        y2="5.6"
      />
      <line
        x1="18"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="16.25"
        x2="18.4"
        y1="16.25"
        y2="18.4"
      />
      <line
        x1="12"
        x2="12"
        y1="18"
        y2="21"
      />
      <line
        x1="7.75"
        x2="5.6"
        y1="16.25"
        y2="18.4"
      />
      <line
        x1="6"
        x2="3"
        y1="12"
        y2="12"
      />
      <line
        x1="7.75"
        x2="5.6"
        y1="7.75"
        y2="5.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;
